File: Data Structures - Hash Tables/Exercise - First Recurring Character/Python/FirstRecurringCharacter.py
def firstRecurringChar(arr):
    recurring_char_map = {}
    for item in arr:
        recurring_char_map[item] = recurring_char_map.get(item, 0) + 1 
        if recurring_char_map[item] > 1:
            return item
    return None

arr = [2,1,1,2,3,5,1,2,4]
result = firstRecurringChar(arr)
print(result)

File: Data Structures - Hash Tables/Exercise - First Recurring Character/JavaScript/FirstRecurringCharacter.js
const firstRecurringChar = (arr) => {
	recurring_char_map = {};
	for (const item of arr) {
		recurring_char_map[item] = (recurring_char_map[item] || 0) + 1;
		if (recurring_char_map[item] > 1)
			return item
	}
	return undefined;
}

const input = [2,3,4,5];
const result = firstRecurringChar(input);
console.log(result);

